src/HDWallet.cpp

```diff
@@ -209,14 +239,24 @@ bool deserialize(const std::string& extended, TWCurve curve, Hash::Hasher hasher
 HDNode getNode(const HDWallet& wallet, TWCurve curve, const DerivationPath& derivationPath) {
     auto node = getMasterNode(wallet, curve);
     for (auto& index : derivationPath.indices) {
-        hdnode_private_ckd(&node, index.derivationIndex());
+        if (curve == TWCurveED25519Extended) {
+            // special handling for Cardano
+            hdnode_private_ckd_cardano(&node, index.derivationIndex());
+        } else {
+            hdnode_private_ckd(&node, index.derivationIndex());
+        }
     }
     return node;
 }
 
 HDNode getMasterNode(const HDWallet& wallet, TWCurve curve) {
     auto node = HDNode();
-    hdnode_from_seed(wallet.seed.data(), HDWallet::seedSize, curveName(curve), &node);
+    if (curve == TWCurveED25519Extended) {
+        // special handling for Cardano, use entropy not seed
```

Comment:
I'm not fan of this solution to associated curve to a specific coin, as `TWCurveED25519Extended ` could be used by some other blockchain and it would use `hdnode_from_seed_cardano`